src-python/drbench/drbench/plotter.py

```diff
@@ -125,25 +135,33 @@ def __stream_strong_scaling_plots(self):
         db_cpu = db.loc[db["device"] == "CPU"]
 
         self.__make_plot(
-            "stream_strong_scaling_cpu",
+            cpu_title,
             db_cpu,
             x="CPU Cores",
             y=Plotter.bandwidth_title,
             col="Benchmark",
             style="Target",
         )
 
-    def __algorithm_plots(self):
+    def __algorithm_plots(self, weak_scaling: bool = False):
         m = self.db_maxvec
 
         db = m.loc[
             m["Benchmark"].isin(["Black_Scholes", "Inclusive_Scan", "Reduce"])
         ].copy()
 
+        if weak_scaling:
+            db = db.loc[db["Weak Scaling"]]
+            gpu_title = "algorithms_gpu_weak_scaling"
```

Comment:
```suggestion
            title = f"{plot}_{device}__{scaling}_scaling"
```

Can you refactor so there is a single function that is passed some combination of "algorithms"|"streams:", "strong"|"weak", "cpu"|"gpu"

and generates the correct plot? You can get a set of data to work with here: https://github.com/oneapi-src/distributed-ranges/suites/15078647182/artifacts/861305619